pkgs/servers/monitoring/cockpit/default.nix

```diff
@@ -0,0 +1,146 @@
+{ lib
+, stdenv
+, fetchzip
+, fetchurl
+, fetchFromGitHub
+, autoreconfHook
+, bash
+, coreutils
+, docbook_xml_dtd_43
+, docbook_xsl
+, findutils
+, gettext
+, git
+, glib
+, glibc
+, gnused
+, gnutls
+, json-glib
+, krb5
+, libssh
+, libxcrypt
+, libxslt
+, makeWrapper
+, nodejs
+, openssl
+, pam
+, pkg-config
+, polkit
+, python3Packages
+, ripgrep
+, systemd
+, udev
+, xmlto
+}:
+stdenv.mkDerivation rec {
+  pname = "cockpit";
+  version = "282.1";
+  src = fetchFromGitHub {
+    owner = "cockpit-project";
+    repo = "cockpit";
+    rev = "83d2ca1295a47c4d2c5a2904a27f8087d2641657";
+    sha256 = "sha256-R3IMga5/j1tmUcXaZpzo3VhgPmsn03yI7aZUnxJouVU=";
+    leaveDotGit = true;
+    fetchSubmodules = true;
+    deepClone = true;
+  };
+  enableParallelBuilding = true;
+  nativeBuildInputs = [
+    autoreconfHook
+    makeWrapper
+
+    docbook_xml_dtd_43
+    docbook_xsl
+    findutils
+    gettext
+    git
+    glib.dev
+    libxcrypt
+    libxslt.bin
+    nodejs
+    pam.out
+    pam.out
+    pkg-config
+    python3Packages.python
+    python3Packages.setuptools
+    ripgrep
+    xmlto
+  ];
+  buildInputs = [
+    glib
+    gnutls
+    json-glib
+    krb5
+    libssh
+    polkit
+    udev
+  ];
+  patches = [
+    ./fix-cockpit-certificate-ensure.patch
+    ./fix-cockpit-certificate-helper.patch
+    ./nerf-node-modules.patch
+  ];
+  configureFlags = [
+    "--enable-prefix-only=yes"
+    "--disable-pcp" # TODO: figure out how to package its dependency
+    "--with-default-session-path=/run/wrappers/bin:/run/current-system/sw/bin"
+  ];
+  patchCodeImpurities = ''
+    patchShebangs tools/escape-to-c
+    patchShebangs tools/make-compile-commands
+    patchShebangs tools/node-modules
+    patchShebangs tools/termschutz
+    patchShebangs tools/webpack-make
+    patchShebangs test/common/tap-cdp
+    patchShebangs test/common/pixel-tests
+    patchShebangs test/common/run-tests
+    export HOME=$(mktemp -d)
+    cp node_modules/.package-lock.json package-lock.json
+    substituteInPlace src/systemd_ctypes/libsystemd.py \
+      --replace libsystemd.so.0 ${systemd}/lib/libsystemd.so.0
+    substituteInPlace pkg/**/*.js pkg/**/*.jsx \
+      --replace '"/usr/bin/' '"' \
+      --replace '"/bin/' '"'
+
+    # for patch testing
+    # cat pkg/users/local.js
+  '';
+  generateVersionFile = ''
+    echo "m4_define(VERSION_NUMBER, [${version}])" > version.m4
+  '';
+  preConfigurePhases = [ "patchCodeImpurities" "generateVersionFile" ];
+  fixupPhase = ''
+    patchShebangs $out/libexec/cockpit-certificate-helper
+    patchShebangs $out/share/cockpit/motd/update-motd
+    PATH=${python3Packages.python.withPackages (p: with p;[ pygobject3 ])}/bin patchShebangs $out/libexec/cockpit-client
+    patchShebangs $out/libexec/cockpit-desktop
+
+    wrapProgram $out/bin/cockpit-bridge \
+      --suffix LD_LIBRARY_PATH : /run/current-system/sw/lib \
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** would this cause impurity?

Reply:
also: trailing backslah?

Reply:
cockpit-bridge is the payload of cockpit

it needs access to the system utilities to do diagnostics and things like that

Reply:
testing rn if it works without this line

Reply:
it's not required, removed the wrapper

Reply:
I added that because at first I packaged it as python library because of the pyproject.toml file but now I realized that the wrapper is not necessary.